src/persist-client/src/impl/machine.rs

```diff
@@ -104,10 +104,6 @@ where
                 Ok(()) => {
                     return Ok(Ok(Ok(seqno)));
                 }
-                // TODO(aljoscha): This upper seems useless because we have to
```

Comment:
I think it's fine to State to return what it does for unit test purposes, so it's fine to machine to then ignore it